arch/x86/cpu/tangier/pinmux.c

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2018 Emlid Limited
+ */
+
+#include <common.h>
+#include <dm.h>
+#include <regmap.h>
+#include <syscon.h>
+#include <asm/cpu.h>
+#include <asm/scu.h>
+#include <linux/io.h>
+
+#define BUFCFG_OFFSET	0x100
+
+#define MRFLD_FAMILY_LEN	0x400
+
+#define pin_to_bufno(f, p)	  ((p) - (f)->pin_base)
+
+struct mrfld_family {
+	unsigned int family_number;
+	unsigned int pin_base;
+	size_t npins;
+	void __iomem *regs;
+};
+
+#define MRFLD_FAMILY(b, s, e)				\
+	{				\
+		.family_number = (b),			\
+		.pin_base = (s),			\
+		.npins = (e) - (s) + 1,			\
+	}
+
+static struct mrfld_family i2c_family = MRFLD_FAMILY(7, 101, 114);
+
+static void __iomem *mrfld_get_bufcfg(struct mrfld_family *family, unsigned int pin)
+{
+	unsigned int bufno;
+
+	bufno = pin_to_bufno(family, pin);
+
+	return family->regs + BUFCFG_OFFSET + bufno * 4;
+}
+
+static void mrfld_setup_families(void *base_addr, struct mrfld_family *families, unsigned int nfam)
+{
+    for (int i = 0; i < nfam; i++) {
+		struct mrfld_family *family = &families[i];
+
+		family->regs = base_addr + family->family_number * MRFLD_FAMILY_LEN;
+	}
+}
+
+int mrfld_pinconfig_protected(unsigned int pin, u32 mask, u32 bits)
+{
+	struct mrfld_family *mf = &i2c_family;
+	int cmd = IPCMSG_INDIRECT_WRITE;
+	void __iomem *bufcfg;
+	u32 v, value;
+	int ret;
+
+	bufcfg = mrfld_get_bufcfg(mf, pin);
+	if (!bufcfg)
+		return -EINVAL;
+
+	value = readl(bufcfg);
+
+	v = (value & ~mask) | (bits & mask);
+
+	debug("scu: cmd: %d v: 0x%x p: 0x%x bits: %d, mask: %d bufcfg: 0x%p\n",
+				cmd, v, (u32) bufcfg, bits, mask, bufcfg);
+
+	ret = scu_ipc_raw_command(cmd, 0, &v, 4, NULL, 0, (u32) bufcfg, 0);
+	if (ret)
+		printf("Failed to set bufcfg via SCU IPC for pin %u (%d)\n", pin, ret);
```

Comment:
Hmm... I don't remember if `err()` stops an execution, please check and use it instead.